scripts/commit_validation/commit_validation/tests/validators/test_crc_validator.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import unittest
+from unittest.mock import patch, mock_open
+
+from commit_validation.tests.mocks.mock_commit import MockCommit
+from commit_validation.validators.crc_validator import CrcValidator
+
+
+class CrcValidatorTests(unittest.TestCase):
+
+    @patch('builtins.open', mock_open(read_data='This file does not contain an AZ_CRC macro'))
+    def test_fileWithNoCrc_passes(self):
+        commit = MockCommit(files=['/someCppFile.cpp'])
+        error_list = []
+        self.assertTrue(CrcValidator().run(commit, error_list))
+        self.assertEqual(len(error_list), 0, f"Unexpected errors: {error_list}")
+
+    @patch('builtins.open', mock_open(read_data='This file contains an invalid CRC macro AZ_CRC("My string", 0xabcdef00)'))
+    def test_fileWithInvalidCrc_fails(self):
+        commit = MockCommit(files=['/someCppFile.cpp'])
+        error_list = []
+        self.assertFalse(CrcValidator().run(commit, error_list))
+        self.assertNotEqual(len(error_list), 0, f"Errors were expected but none were returned.")
+
+    @patch('builtins.open', mock_open(read_data='This file contains a valid CRC macro AZ_CRC("My string", 0x18fbd270)'))
+    def test_fileWithValidCrc_fails(self):
+        commit = MockCommit(files=['/someCppFile.cpp'])
+        error_list = []
+        self.assertTrue(CrcValidator().run(commit, error_list))
+        self.assertEqual(len(error_list), 0, f"Unexpected errors: {error_list}")
+
+    @patch('builtins.open', mock_open(read_data='This file contains an invalid CRC macro AZ_CRC("My string", 0xabcdef00)'))
+    def test_fileExtensionIgnored_passes(self):
+        commit = MockCommit(files=['/someCppFile.somerandomextension'])
+        error_list = []
+        self.assertTrue(CrcValidator().run(commit, error_list))
+        self.assertEqual(len(error_list), 0, f"Unexpected errors: {error_list}")
+
+if __name__ == '__main__':
```

Comment:
You should add this to CTest using the `ly_add_pytest` function

Reply:
I though the registration was here: `scripts\commit_validation\CMakeLists.txt` but looks like is pointing to the wrong directory, will fix and register these too

Reply:
So it is scanning for test in the entire `scripts/commit_validation` directory

Reply:
yeah, added the `/tests` to just scan the test folder